KarhooUISDK/Screens/CheckoutScreen/NewCheckout/Workers/KarhooNewCheckoutBookingWorker.swift

```diff
@@ -191,4 +336,13 @@ final class KarhooNewCheckoutBookingWorker: NewCheckoutBookingWorker {
                 //            self.bookingState = .failure(ErrorModel(message: "something_went_wrong", code: "0231"))
             )})
     }
+
+    private func isLoyaltyEnabled() -> Bool {
+        let loyaltyId = userService.getCurrentUser()?.paymentProvider?.loyaltyProgamme.id
+        return loyaltyId != nil && !loyaltyId!.isEmpty && LoyaltyFeatureFlags.loyaltyEnabled
+    }
+
+    private func isKarhooUser() -> Bool {
+        sdkConfiguration.authenticationMethod().isGuest() == false
```

Comment:
on old KarchooCheckutPresenter value `isKarhooUser` war defined as: `sdkConfiguration.authenticationMethod() == .karhooUser`, and it doesn't mean the same, because in your implementation bookingExchange return also true, in old implementation not. Are you sure you wanted change this?

Reply:
Good spotting @BartlomiejSopalaKarhoo. We need to go with the old comparison here.